document/CoreFormat.mdk

```diff
@@ -355,12 +359,52 @@ An `issue` object describes a single issue detected by a static analysis tool.
 ### `ruleId` property { #issue-ruleId }
 
 An `issue` object shall contain a property named `ruleId` whose value is a string containing
-a stable, opaque identifier for the rule that was violated.
+the stable, opaque identifier for the rule that was violated.
 
 ~ IsoExample
 `"CA2101"`
 ~
 
+### `kind` property { #issue-kind }
+
+An `issue` object may contain a property named `kind` whose value is one of a fixed set of strings
+that specify what kind of issue this is.
+
+If present, the `kind` property shall have one of the following values, with the specified meanings:
+
+* `warning`: The rule specified by the `ruleId` property (&sect;[#issue-ruleId]) was evaluated and a violation was found.
+
+* `error`: The rule specified by the `ruleId` property was evaluated and a serious violation was found.
+
+* `pending`: The rule specified by the `ruleId` property was evaluated, but whether the result of the evaluation
+constitutes a violation cannot be determined without further evaluation, either by a person or by an automated tool.
+
+~ IsoNote
+In some cases, team policy or corporate policy might determine if an analysis result constitutes a violation.
+~
+
+* `internalError`: The tool encountered an internal error that prevented it from performing a complete analysis.
+
+    If the error occurred in the course of evaluating a specific rule, the `ruleId` property should specify that rule.
```

Comment:
this is an interesting point. current tools define a specific id for 'unhandled rule exception'. I don't think we want to specify any behavior here re: the rule id, as this gets into the domain of how to implement checkers rather than defining a structure for persisting results. 

if you specify a unique id, you can easily filter log files for unhandled rule exceptions by querying for that id. I see your point, though, that since we have a rule id, why not provide it? current tools emit it to the full message